src/middlewares/authorization.js

```diff
@@ -0,0 +1,24 @@
+import decodeJwt from '../helpers/tokens/decode.token';
+import sendError from '../helpers/error.sender';
+import errorMessages from '../helpers/constants/error.messages';
+import status from '../helpers/constants/status.codes';
+
+const payload = (req, res, next) => {
+  // initializing variables
+  let tempUser;
+
+  try {
```

Comment:
why 2 blocks of try catch??

Reply:
it's catching two different cases of errors. 
Try to see the logic @alainburindi

Reply:
Why cant you use only one for both cases?

Reply:
Because I will be losing one case. But, after @murediane 's PR will be merged, I will use her authorization middleware instead of this